python/context/ball_trajectories.py

```diff
@@ -274,6 +306,233 @@ def overwrite(
         traj_group.create_dataset(self._TIME_STAMPS, data=time_stamps)
         traj_group.create_dataset(self._TRAJECTORY, data=positions)
 
+    def add_ball_robot_trajectories(
+        self, group_name: str, o80_robot_ball_path: pathlib.Path
+    ) -> int:
+        """
+        It is assumed that o80_robot_ball_path is a directory hosting
+        a collection of files named o80_robot_ball_* that have been generated
+        by the executable o80_robot_ball_logger (package o80_pam). This function
+        will parse all these files and add them to the hdf5 under the specified
+        group name.
+
+        Returns
+        -------
+        The number of trajectories added to the file.
+        """
+
+        def _read_trajectory(o80_robot_ball_file: pathlib.Path) -> StampedTrajectory:
```

Comment:
The amount of nested functions here makes it a bit difficult to follow the code.  Maybe it would make sense to move them to a separate module?